### ISSUE TITLE
between predicate - nulls

### ISSUE THREAD

**Issue by mbutlerw** [xtdb/crux#2429]
I believe due to the rules around nulls in comparison predicates, that is if either operand is null the result is null, predicates than contain divide by zero etc. should short circuit and return null, rather than trying to run the divide by zero and erroring.

I think we already do this for simple comparison preds e.g.

Current issue we define between in terms of AND, which doesn't respect the nulls shorting flag. Even if it did, we perhaps need a way to explicitly pass down "short circuit nulls for this specific expr" as we have a number of exprs defined as lower level exprs.
 
```SELECT tab1.col2 FROM tab1 WHERE NULL = tab1.col0 / 0```

For example, this query should not error

```
SELECT tab1.col2 FROM tab1 WHERE tab1.col0 BETWEEN NULL AND tab1.col0 / 0

=>

[:project
 [x3]
 [:rename
  {col0 x1, col1 x2, col2 x3, _table x4}
  [:scan
   [{col0 (between col0 nil (/ col0 0))}
    col1
    col2
    {_table (= _table "tab1")}]]]]

**Comment by jarohen** [comment]
Discussed offline: we think the spec mandates that `x BETWEEN NULL and <error>` should error - `BETWEEN` expands into an `AND`, which does not (fully) short-circuit when the LHS is NULL (the RHS may still be false). We compared `SELECT x BETWEEN ...` with `SELECT ... WHERE x BETWEEN ...` - it seems that some other databases are doing other optimisations when the `BETWEEN` is part of the `WHERE` clause (because this collapses to 2VL) vs when it's in the `SELECT` (3VL), although SQLite seems to still return null in the latter case (which we believe to be incorrect).

tl;dr we're happy enough with our behaviour to mark these with a `skipif` in SLT.
